network/connection.go

```diff
@@ -198,6 +198,16 @@ func (conn *Connection) Deny(reason string) {
 	}
 }
 
+// Failed marks the connection with VerdictFailed and stores the reason.
+func (conn *Connection) Failed(reason string) {
+	if conn.SetVerdict(VerdictFailed) {
+		conn.Reason = reason
+		log.Infof("filter: dropping connection %s because of an internal error: %s", conn, reason)
+	} else {
+		log.Warningf("filter: tried to drop %s due to error but current verdict is %s", conn, conn.Verdict)
+	}
+}
```

Comment:
`conn.SetVerdict` will always succeed here, as `VerdictFailed` is the highest numerical verdict. This is good and intended, because we definitely want to drop if we are unable to find a Verdict without error.
We can leave this here, as you wish.

Reply:
Ah interesting, didn't check how `SetVerdict` actually works. But yeah, I guess we can leave it in there so any future verdict (with a highter number) won't cause problems here.